### PR TITLE
make sure CMSSW always finds HOME in environment, fixes #6894

### LINKED ISSUE #6894: Exit Code 134 $HOME/.root.mimes
the frequent error code 134 seems to be related to  $HOME/.root.mimes

does wmcore make reference to this file, how does this get accessed by the job ?

https://hypernews.cern.ch/HyperNews/CMS/get/edmFramework/3654.html

this is causing major failures and inefficiencies in production.

### PR DESCRIPTION
**Pull request by hufnagel**
Progression in environment variable checks is based on what I observed in Tier0 job environments.

Tested the more generic code changes, but obviously can't test the 'HOME not set use case' since the Tier0 jobs have HOME set.

@amaltaro @vlimant please review and maybe try out in a production agent to see what it does. Can't be worse than CMSSW crashing because HOME not being set.

### CODE REVIEW

**Comment by amaltaro** [comment]
Where this issue come from? Do we have a GH issue for that? Why would CMSSW not find $HOME variable in a few ... worker nodes but succeed in others?

If you can give me more details on where it failed, which workflow or configuration, we can setup it in *testbed* for testing.

**Comment by hufnagel** [comment]
Check #6894 and the HN discussion. I just wrote the patch, @vlimant should follow up.

**Comment by vlimant** [comment]
Hi Dirk, that's nice to force set it as an immediate fix.
I think we have to follow up with glidein to understand how it does not get set systematically.
Furthermore, we should find a way to report "some text" about the non-CMSSW exception (like that one, and segfault, and all) that result in 134, so that we can under the actual origin without digging up a log

**Comment by hufnagel** [comment]
So it seems like it is set in a frontend validation script.

https://github.com/jamesletts/CMSglideinWMSValidation/blob/master/set_home_cms.source

Given this, should I simplify this patch look only at PWD with a fallback to / maybe as a safety feature ?

**Comment by amaltaro** [comment]
Please do so, I feel like crying when I look at that line 230 :)

**Comment by vlimant** [comment]
@amaltaro you don't have a taste for russian dolls ?

File: src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py [hunk 1]
```diff
@@ -218,11 +218,20 @@ def execute(self, emulator=None):
                 cmsswArguments]
         logging.info("Executing CMSSW. args: %s", args)
 
-        # at CERN override the environment to work around problem with GSI authentication plugin and EOS
+        #
+        # possibly needed environment overrides for for CMSSW call go here
+        #
```
Comment (amaltaro) [inline src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py]: Can you reduce these 3 commented lines into 1 only? "for for" btw

**Comment by amaltaro** [comment]
@vlimant since this is going to be a tricky validation, can you point me to the site where most of these errors came from? I'll send 2k jobs there and see whether we hit this 134 exit code again.

Moreover, I understood it depends on the library being used in CMSSW(?) So can you also point me the workflow name?

**Comment by vlimant** [comment]
@amaltaro https://cms-logbook.cern.ch/elog/Workflow+processing/24500 pdmvserv_SUS-RunIISpring16FSPremix-00009_00008_v0__160526_002050_6726 ran on the HLT (admittedly using overflowing)

**Comment by amaltaro** [comment]
Thanks, failures both at CERN and CERN_HLT. I'll clone this worfklow later on and run some jobs for the weekend.

**Comment by amaltaro** [comment]
The clone workflow is about to finish, about 2/3 of the jobs failed but there was not a single exit code 134. I think we can consider this fix as done.

**Comment by amaltaro** [comment]
Merging it now such that we can cut a new stable release.

**Pull request merged**